Bump protobuf-java and protobuf-java-util from 4.31.1 to 4.33.1

nico-semko opened the pull request:
Restores backward compatibility with protobuf v3/v4 generated code. This matches the upgrade done in rules_kotlin to address compatibility issues with Kotlin annotation processors like KSP that need to load protobufs generated with both v3 and v4 protoc.

Related change to `rules_kotlin` : https://github.com/bazelbuild/rules_kotlin/pull/1413

Currently `rules_kotlin` has a version conflict that could be solved by downgrading the lower version https://github.com/bazelbuild/rules_kotlin/pull/1429 but that will revert restored backward compatibility with protobuf v3/v4 generated code that is available in 4.32.+. Let's use the latest available version of libs instead.

## Problem
JavaProto 4.x previously dropped compatibility with potentially vulnerable generated code (pre-3.22.x gencode impacted by https://github.com/advisories/GHSA-h4h5-3hr4-j3g2). This caused issues for clients using Kotlin annotation processors like KSP that need to load protobufs generated with both v3 and v4 protoc.

## Solution
Upgrade to protobuf-java 4.33.1, which includes the compatibility restoration introduced in 4.32.0+ ([commit 7c51e5b](https://github.com/protocolbuffers/protobuf/commit/7c51e5b582f098987b6fda8d590c95e9b5d53fd7)).


## Changes:
- Updated protobuf Bazel module version from 31.1 to 33.1
- Updated protobuf-java from 4.31.1 to 4.33.1
- Updated protobuf-java-util from 4.31.1 to 4.33.1

## References:
- https://github.com/protocolbuffers/protobuf/releases/tag/v33.1
- https://github.com/protocolbuffers/protobuf/releases/tag/v33.0
- https://protobuf.dev/support/cross-version-runtime-guarantee/#backwards

nico-semko committed:
Bump protobuf-java and protobuf-java-util from 4.31.1 to 4.33.1

Restores backward compatibility with protobuf v3/v4 generated code.
This matches the upgrade done in rules_kotlin to address compatibility
issues with Kotlin annotation processors like KSP that need to load
protobufs generated with both v3 and v4 protoc.

Changes:
- Updated protobuf Bazel module version from 31.1 to 33.1
- Updated protobuf-java from 4.31.1 to 4.33.1
- Updated protobuf-java-util from 4.31.1 to 4.33.1
- Added proto_version variable to keep versions in sync

References:
- https://github.com/protocolbuffers/protobuf/releases/tag/v33.1
- https://github.com/protocolbuffers/protobuf/releases/tag/v33.0
- https://protobuf.dev/support/cross-version-runtime-guarantee/#backwards

Wyverald commented on java/rules_jvm_external++maven+bazel_worker_maven_install.json:
what is this file? it doesn't seem to be used.

nico-semko replied:
The rules_jvm_external++maven+bazel_worker_maven_install.json file was generated when repinning Maven dependencies, `maven.install("bazel_worker_maven", ...)` from `java/MODULE.bazel`.

 I will revert this change as this file is not used. Thanks

Wyverald commented on java/MODULE.bazel:
```diff
@@ -3,9 +3,15 @@ module(
     version = "0",
 )
 
+proto_version = "4.33.1"
```
can we just inline these? I think a comment saying "keep in sync with with protobuf-java" or some such would be more straightforward and still fairly foolproof.

nico-semko replied:
Sure, I could inline + comment.

nico-semko committed:
Add pin file to .gitignore

The rules_jvm_external++maven+bazel_worker_maven_install.json file is
auto-generated by Bazel's bzlmod system and doesn't need to be checked in.

nico-semko reviewed (commented):
Changed

Bencodes commented:
@Wyverald can you cut a new release with these changes?
